Harfbuzz 2.6.1 - Can't be built on Mac OS with CMake

Hi!

Thanks for release a new version of Harfbuzz!

When building on Mac Os + clang I have the follow error:

```
----Running------
> cd '/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/build_subfolder' && cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE="Release" -DCONAN_EXPORTED="1" -DCONAN_IN_LOCAL_CACHE="ON" -DCONAN_COMPILER="apple-clang" -DCONAN_COMPILER_VERSION="10.0" -DCONAN_CXX_FLAGS="-m64" -DCONAN_SHARED_LINKER_FLAGS="-m64" -DCONAN_C_FLAGS="-m64" -DBUILD_SHARED_LIBS="OFF" -DCMAKE_INSTALL_PREFIX="/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/package/693541ba016479648be7388be09335abae7da778" -DCMAKE_INSTALL_BINDIR="bin" -DCMAKE_INSTALL_SBINDIR="bin" -DCMAKE_INSTALL_LIBEXECDIR="bin" -DCMAKE_INSTALL_LIBDIR="lib" -DCMAKE_INSTALL_INCLUDEDIR="include" -DCMAKE_INSTALL_OLDINCLUDEDIR="include" -DCMAKE_INSTALL_DATAROOTDIR="share" -DCONAN_CMAKE_POSITION_INDEPENDENT_CODE="ON" -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY="ON" -DCMAKE_C_FLAGS="-Wno-deprecated-declarations" -DCMAKE_CXX_FLAGS="-Wno-deprecated-declarations" -DCMAKE_MACOSX_RPATH="True" -DHB_HAVE_FREETYPE="True" -DHB_BUILD_TESTS="False" -DHB_HAVE_ICU="False" -DHB_HAVE_GLIB="True" -Wno-dev '/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778'
-----------------
-- The C compiler identification is AppleClang 10.0.1.10010046
-- The CXX compiler identification is AppleClang 10.0.1.10010046
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: called by CMake conan helper
-- Conan: called inside local cache
-- Conan: Adjusting output directories
-- Conan: Using cmake global configuration
-- Conan: Adjusting default RPATHs Conan policies
-- Conan: Adjusting language standard
-- Conan: Adjusting fPIC flag (ON)
-- Found PythonInterp: /Users/user914788/.pyenv/versions/3.7.1/bin/python (found version "3.7.1") 
-- Looking for atexit
-- Looking for atexit - found
-- Looking for mprotect
-- Looking for mprotect - found
-- Looking for sysconf
-- Looking for sysconf - found
-- Looking for getpagesize
-- Looking for getpagesize - found
-- Looking for mmap
-- Looking for mmap - found
-- Looking for isatty
-- Looking for isatty - found
-- Looking for newlocale
-- Looking for newlocale - found
-- Looking for strtod_l
-- Looking for strtod_l - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Looking for sys/mman.h
-- Looking for sys/mman.h - found
-- Looking for xlocale.h
-- Looking for xlocale.h - found
-- Looking for stdbool.h
-- Looking for stdbool.h - found
-- Found Freetype: /Users/user914788/.conan/data/freetype/2.10.0/bincrafters/stable/package/cb656f6d884f8fc76ab61dd32ede672f98113876/lib/libfreetype.a (found version "2.10.0") 
-- Looking for FT_Get_Var_Blend_Coordinates
-- Looking for FT_Get_Var_Blend_Coordinates - found
-- Looking for FT_Set_Var_Blend_Coordinates
-- Looking for FT_Set_Var_Blend_Coordinates - found
-- Looking for FT_Done_MM_Var
-- Looking for FT_Done_MM_Var - found
-- Found PkgConfig: /usr/local/bin/pkg-config (found version "0.29.2") 
-- Performing Test CXX_SUPPORTS_FLAG_BSYMB_FUNCS
-- Performing Test CXX_SUPPORTS_FLAG_BSYMB_FUNCS - Success
-- Performing Test COMPILER_SUPPORTS_CXX11
-- Performing Test COMPILER_SUPPORTS_CXX11 - Success
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY


-- Build files have been written to: /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/build_subfolder

----Running------
> cmake --build '/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/build_subfolder' '--' '-j8'
-----------------
/usr/local/Cellar/cmake/3.15.0/bin/cmake -S/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778 -B/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/build_subfolder --check-build-system CMakeFiles/Makefile.cmake 0
/usr/local/Cellar/cmake/3.15.0/bin/cmake -E cmake_progress_start /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/build_subfolder/CMakeFiles /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/build_subfolder/CMakeFiles/progress.marks
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/Makefile2 all
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f source_subfolder/CMakeFiles/harfbuzz.dir/build.make source_subfolder/CMakeFiles/harfbuzz.dir/depend
cd /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/build_subfolder && /usr/local/Cellar/cmake/3.15.0/bin/cmake -E cmake_depends "Unix Makefiles" /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778 /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/build_subfolder /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/build_subfolder/source_subfolder /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/build_subfolder/source_subfolder/CMakeFiles/harfbuzz.dir/DependInfo.cmake --color=
Scanning dependencies of target harfbuzz
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f source_subfolder/CMakeFiles/harfbuzz.dir/build.make source_subfolder/CMakeFiles/harfbuzz.dir/build
[  8%] Building CXX object source_subfolder/CMakeFiles/harfbuzz.dir/src/harfbuzz.cc.o
cd /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/build_subfolder/source_subfolder && /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -DHAVE_ATEXIT -DHAVE_CORETEXT -DHAVE_FREETYPE=1 -DHAVE_FT_DONE_MM_VAR -DHAVE_FT_GET_VAR_BLEND_COORDINATES -DHAVE_FT_SET_VAR_BLEND_COORDINATES -DHAVE_GETPAGESIZE -DHAVE_GLIB -DHAVE_INTEL_ATOMIC_PRIMITIVES -DHAVE_ISATTY -DHAVE_MMAP -DHAVE_MPROTECT -DHAVE_NEWLOCALE -DHAVE_STDBOOL_H -DHAVE_STRTOD_L -DHAVE_SYSCONF -DHAVE_SYS_MMAN_H -DHAVE_UNISTD_H -DHAVE_XLOCALE_H -I/Users/user914788/.conan/data/freetype/2.10.0/bincrafters/stable/package/cb656f6d884f8fc76ab61dd32ede672f98113876/include -I/Users/user914788/.conan/data/freetype/2.10.0/bincrafters/stable/package/cb656f6d884f8fc76ab61dd32ede672f98113876/include/freetype2 -I/Users/user914788/.conan/data/glib/2.58.3/bincrafters/stable/package/3211aa20c3b4b8be9ac16650df4ce4ee34cee278/include -I/Users/user914788/.conan/data/glib/2.58.3/bincrafters/stable/package/3211aa20c3b4b8be9ac16650df4ce4ee34cee278/include/glib-2.0 -I/Users/user914788/.conan/data/glib/2.58.3/bincrafters/stable/package/3211aa20c3b4b8be9ac16650df4ce4ee34cee278/lib/glib-2.0/include -I/Users/user914788/.conan/data/libpng/1.6.37/bincrafters/stable/package/9f67cefb3b597b342ffd9241a4a7dc5a0df50bef/include -I/Users/user914788/.conan/data/libffi/3.2.1/bincrafters/stable/package/534dcc368c999e07e81f146b3466b8f656ef1f55/lib/libffi-3.2.1/include -I/Users/user914788/.conan/data/pcre/8.41/bincrafters/stable/package/fc3175c1863d25e78970d6201cdaf9c2a294f88a/include -I/Users/user914788/.conan/data/libelf/0.8.13/bincrafters/stable/package/bad7dd1e4b5daf2b9ae8e3ca2e4397d55b8d3583/include -I/Users/user914788/.conan/data/gettext/0.20.1/bincrafters/stable/package/6c3a931daae9d7916b13c01550e469037fec642b/include -I/Users/user914788/.conan/data/bzip2/1.0.8/conan/stable/package/25d69afe851bfb8cb5aedd20123fa41e061f316e/include -I/Users/user914788/.conan/data/libxml2/2.9.9/bincrafters/stable/package/f4d4dc389913ba7d6be3b026feb6e93a89b39006/include/libxml2 -I/Users/user914788/.conan/data/zlib/1.2.11/conan/stable/package/534dcc368c999e07e81f146b3466b8f656ef1f55/include -I/Users/user914788/.conan/data/libiconv/1.15/bincrafters/stable/package/534dcc368c999e07e81f146b3466b8f656ef1f55/include -I/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src -I/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/build_subfolder/source_subfolder/src -I/Library/Frameworks/Mono.framework/Headers  -Wno-deprecated-declarations  -m64 -fno-rtti -fno-exceptions -fno-threadsafe-statics -std=c++11 -O3 -DNDEBUG  -fPIC   -DLIBXML_STATIC -DPCRE_STATIC=1 -DFFI_STATIC -o CMakeFiles/harfbuzz.dir/src/harfbuzz.cc.o -c /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/harfbuzz.cc
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/harfbuzz.cc:53:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.cc:35:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.h:37:
In file included from /System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:
In file included from /System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:19:
In file included from /System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:43:
In file included from /System/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:77:
/usr/include/MacTypes.h:194:9: error: reference to 'Fixed' is ambiguous
typedef Fixed *                         FixedPtr;
        ^
/usr/include/MacTypes.h:193:41: note: candidate found by name lookup is 'Fixed'
typedef SInt32                          Fixed;
                                        ^
/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-open-type.hh:119:8: note: candidate found by name lookup is 'OT::Fixed'
struct Fixed : HBINT32
       ^
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/harfbuzz.cc:53:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.cc:35:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.h:37:
In file included from /System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:
In file included from /System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:19:
In file included from /System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:43:
In file included from /System/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:77:
/usr/include/MacTypes.h:553:3: error: reference to 'Fixed' is ambiguous
  Fixed               x;
  ^
/usr/include/MacTypes.h:193:41: note: candidate found by name lookup is 'Fixed'
typedef SInt32                          Fixed;
                                        ^
/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-open-type.hh:119:8: note: candidate found by name lookup is 'OT::Fixed'
struct Fixed : HBINT32
       ^
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/harfbuzz.cc:53:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.cc:35:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.h:37:
In file included from /System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:
In file included from /System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:19:
In file included from /System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:43:
In file included from /System/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:77:
/usr/include/MacTypes.h:554:3: error: reference to 'Fixed' is ambiguous
  Fixed               y;
  ^
/usr/include/MacTypes.h:193:41: note: candidate found by name lookup is 'Fixed'
typedef SInt32                          Fixed;
                                        ^
/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-open-type.hh:119:8: note: candidate found by name lookup is 'OT::Fixed'
struct Fixed : HBINT32
       ^
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/harfbuzz.cc:53:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.cc:35:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.h:37:
In file included from /System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:
In file included from /System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:19:
In file included from /System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:43:
In file included from /System/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:77:
/usr/include/MacTypes.h:558:3: error: reference to 'Fixed' is ambiguous
  Fixed               left;
  ^
/usr/include/MacTypes.h:193:41: note: candidate found by name lookup is 'Fixed'
typedef SInt32                          Fixed;
                                        ^
/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-open-type.hh:119:8: note: candidate found by name lookup is 'OT::Fixed'
struct Fixed : HBINT32
       ^
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/harfbuzz.cc:53:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.cc:35:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.h:37:
In file included from /System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:
In file included from /System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:19:
In file included from /System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:43:
In file included from /System/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:77:
/usr/include/MacTypes.h:559:3: error: reference to 'Fixed' is ambiguous
  Fixed               top;
  ^
/usr/include/MacTypes.h:193:41: note: candidate found by name lookup is 'Fixed'
typedef SInt32                          Fixed;
                                        ^
/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-open-type.hh:119:8: note: candidate found by name lookup is 'OT::Fixed'
struct Fixed : HBINT32
       ^
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/harfbuzz.cc:53:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.cc:35:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.h:37:
In file included from /System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:
In file included from /System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:19:
In file included from /System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:43:
In file included from /System/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:77:
/usr/include/MacTypes.h:560:3: error: reference to 'Fixed' is ambiguous
  Fixed               right;
  ^
/usr/include/MacTypes.h:193:41: note: candidate found by name lookup is 'Fixed'
typedef SInt32                          Fixed;
                                        ^
/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-open-type.hh:119:8: note: candidate found by name lookup is 'OT::Fixed'
struct Fixed : HBINT32
       ^
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/harfbuzz.cc:53:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.cc:35:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.h:37:
In file included from /System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:
In file included from /System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:19:
In file included from /System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:43:
In file included from /System/Library/Frameworks/CoreFoundation.framework/Headers/CFBase.h:77:
/usr/include/MacTypes.h:561:3: error: reference to 'Fixed' is ambiguous
  Fixed               bottom;
  ^
/usr/include/MacTypes.h:193:41: note: candidate found by name lookup is 'Fixed'
typedef SInt32                          Fixed;
                                        ^
/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-open-type.hh:119:8: note: candidate found by name lookup is 'OT::Fixed'
struct Fixed : HBINT32
       ^
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/harfbuzz.cc:53:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.cc:35:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.h:37:
In file included from /System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:
In file included from /System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:19:
In file included from /System/Library/Frameworks/CoreFoundation.framework/Headers/CoreFoundation.h:90:
In file included from /System/Library/Frameworks/CoreFoundation.framework/Headers/CFFileSecurity.h:16:
/usr/include/sys/acl.h:154:72: error: expected ')'
extern int      acl_create_entry_np(acl_t *acl_p, acl_entry_t *entry_p, int entry_index);
                                                                            ^
/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-ot-name-table.hh:39:24: note: expanded from macro 'entry_index'
#define entry_index var.u16[1]
                       ^
/usr/include/sys/acl.h:154:31: note: to match this '('
extern int      acl_create_entry_np(acl_t *acl_p, acl_entry_t *entry_p, int entry_index);
                                   ^
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/harfbuzz.cc:53:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.cc:35:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.h:37:
In file included from /System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:
In file included from /System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:24:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/AE.framework/Headers/AE.h:20:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/CarbonCore.h:35:
/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/FixMath.h:123:8: error: reference to 'Fixed' is ambiguous
extern Fixed 
       ^
/usr/include/MacTypes.h:193:41: note: candidate found by name lookup is 'Fixed'
typedef SInt32                          Fixed;
                                        ^
/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-open-type.hh:119:8: note: candidate found by name lookup is 'OT::Fixed'
struct Fixed : HBINT32
       ^
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/harfbuzz.cc:53:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.cc:35:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.h:37:
In file included from /System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:
In file included from /System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:24:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/AE.framework/Headers/AE.h:20:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/CarbonCore.h:35:
/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/FixMath.h:137:8: error: reference to 'Fixed' is ambiguous
extern Fixed 
       ^
/usr/include/MacTypes.h:193:41: note: candidate found by name lookup is 'Fixed'
typedef SInt32                          Fixed;
                                        ^
/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-open-type.hh:119:8: note: candidate found by name lookup is 'OT::Fixed'
struct Fixed : HBINT32
       ^
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/harfbuzz.cc:53:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.cc:35:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.h:37:
In file included from /System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:
In file included from /System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:24:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/AE.framework/Headers/AE.h:20:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/CarbonCore.h:35:
/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/FixMath.h:139:3: error: reference to 'Fixed' is ambiguous
  Fixed   a,
  ^
/usr/include/MacTypes.h:193:41: note: candidate found by name lookup is 'Fixed'
typedef SInt32                          Fixed;
                                        ^
/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-open-type.hh:119:8: note: candidate found by name lookup is 'OT::Fixed'
struct Fixed : HBINT32
       ^
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/harfbuzz.cc:53:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.cc:35:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.h:37:
In file included from /System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:
In file included from /System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:24:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/AE.framework/Headers/AE.h:20:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/CarbonCore.h:35:
/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/FixMath.h:140:3: error: reference to 'Fixed' is ambiguous
  Fixed   b)                                                  __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
  ^
/usr/include/MacTypes.h:193:41: note: candidate found by name lookup is 'Fixed'
typedef SInt32                          Fixed;
                                        ^
/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-open-type.hh:119:8: note: candidate found by name lookup is 'OT::Fixed'
struct Fixed : HBINT32
       ^
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/harfbuzz.cc:53:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.cc:35:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.h:37:
In file included from /System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:
In file included from /System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:24:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/AE.framework/Headers/AE.h:20:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/CarbonCore.h:35:
/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/FixMath.h:152:10: error: reference to 'Fixed' is ambiguous
FixRound(Fixed x)                                             __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
         ^
/usr/include/MacTypes.h:193:41: note: candidate found by name lookup is 'Fixed'
typedef SInt32                          Fixed;
                                        ^
/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-open-type.hh:119:8: note: candidate found by name lookup is 'OT::Fixed'
struct Fixed : HBINT32
       ^
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/harfbuzz.cc:53:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.cc:35:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.h:37:
In file included from /System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:
In file included from /System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:24:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/AE.framework/Headers/AE.h:20:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/CarbonCore.h:35:
/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/FixMath.h:164:10: error: reference to 'Fixed' is ambiguous
Fix2Frac(Fixed x)                                             __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
         ^
/usr/include/MacTypes.h:193:41: note: candidate found by name lookup is 'Fixed'
typedef SInt32                          Fixed;
                                        ^
/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-open-type.hh:119:8: note: candidate found by name lookup is 'OT::Fixed'
struct Fixed : HBINT32
       ^
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/harfbuzz.cc:53:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.cc:35:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.h:37:
In file included from /System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:
In file included from /System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:24:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/AE.framework/Headers/AE.h:20:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/CarbonCore.h:35:
/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/FixMath.h:176:10: error: reference to 'Fixed' is ambiguous
Fix2Long(Fixed x)                                             __OSX_AVAILABLE_BUT_DEPRECATED(__MAC_10_0, __MAC_10_8, __IPHONE_NA, __IPHONE_NA);
         ^
/usr/include/MacTypes.h:193:41: note: candidate found by name lookup is 'Fixed'
typedef SInt32                          Fixed;
                                        ^
/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-open-type.hh:119:8: note: candidate found by name lookup is 'OT::Fixed'
struct Fixed : HBINT32
       ^
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/harfbuzz.cc:53:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.cc:35:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.h:37:
In file included from /System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:
In file included from /System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:24:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/AE.framework/Headers/AE.h:20:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/CarbonCore.h:35:
/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/FixMath.h:187:8: error: reference to 'Fixed' is ambiguous
extern Fixed 
       ^
/usr/include/MacTypes.h:193:41: note: candidate found by name lookup is 'Fixed'
typedef SInt32                          Fixed;
                                        ^
/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-open-type.hh:119:8: note: candidate found by name lookup is 'OT::Fixed'
struct Fixed : HBINT32
       ^
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/harfbuzz.cc:53:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.cc:35:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.h:37:
In file included from /System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:
In file included from /System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:24:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/AE.framework/Headers/AE.h:20:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/CarbonCore.h:35:
/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/FixMath.h:199:8: error: reference to 'Fixed' is ambiguous
extern Fixed 
       ^
/usr/include/MacTypes.h:193:41: note: candidate found by name lookup is 'Fixed'
typedef SInt32                          Fixed;
                                        ^
/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-open-type.hh:119:8: note: candidate found by name lookup is 'OT::Fixed'
struct Fixed : HBINT32
       ^
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/harfbuzz.cc:53:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.cc:35:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.h:37:
In file included from /System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:
In file included from /System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:24:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/AE.framework/Headers/AE.h:20:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/CarbonCore.h:35:
/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/FixMath.h:225:8: error: reference to 'Fixed' is ambiguous
extern Fixed 
       ^
/usr/include/MacTypes.h:193:41: note: candidate found by name lookup is 'Fixed'
typedef SInt32                          Fixed;
                                        ^
/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-open-type.hh:119:8: note: candidate found by name lookup is 'OT::Fixed'
struct Fixed : HBINT32
       ^
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/harfbuzz.cc:53:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.cc:35:
In file included from /Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-coretext.h:37:
In file included from /System/Library/Frameworks/ApplicationServices.framework/Headers/ApplicationServices.h:23:
In file included from /System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:24:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/AE.framework/Headers/AE.h:20:
In file included from /System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/CarbonCore.h:35:
/System/Library/Frameworks/CoreServices.framework/Frameworks/CarbonCore.framework/Headers/FixMath.h:227:3: error: reference to 'Fixed' is ambiguous
  Fixed   x,
  ^
/usr/include/MacTypes.h:193:41: note: candidate found by name lookup is 'Fixed'
typedef SInt32                          Fixed;
                                        ^
/Users/user914788/.conan/data/harfbuzz/2.6.1/bincrafters/testing/build/693541ba016479648be7388be09335abae7da778/source_subfolder/src/hb-open-type.hh:119:8: note: candidate found by name lookup is 'OT::Fixed'
struct Fixed : HBINT32
       ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[2]: *** [source_subfolder/CMakeFiles/harfbuzz.dir/src/harfbuzz.cc.o] Error 1
make[1]: *** [source_subfolder/CMakeFiles/harfbuzz.dir/all] Error 2
make: *** [all] Error 2
```

As you can see, `reference to 'Fixed' is ambiguous` is related to OSX CoreServices.

#### Environment

- System Version: macOS 10.14.6 (18G95)
- Kernel Version: Darwin 18.7.0
- Apple LLVM version 10.0.1 (clang-1001.0.46.4)

#### Steps to reproduce

    wget https://github.com/harfbuzz/harfbuzz/archive/2.6.1.tar.gz
    tar zxf 2.6.1.tar.gz
    cd harfbuzz-2.6.1
    mkdir build
    cd build
    cmake .. -DCMAKE_BUILD_TYPE=Release
    cmake --build .

Of course, running `autogen.sh` it works, but it's not what I expect since from **2.4.0** I'm totally able to build using CMake only.

Regards!

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -27,7 +27,7 @@ jobs:
       xcode: "11.0.0"
     steps:
       - checkout
-      - run: HOMEBREW_NO_AUTO_UPDATE=1 brew install wget autoconf automake libtool pkg-config ragel freetype glib cairo icu4c graphite2
+      - run: HOMEBREW_NO_AUTO_UPDATE=1 brew install wget autoconf automake libtool pkg-config ragel freetype glib cairo icu4c graphite2 cmake
       - run: export PKG_CONFIG_PATH="/usr/local/opt/icu4c/lib/pkgconfig:/usr/local/opt/libffi/lib/pkgconfig" && ./autogen.sh --with-freetype --with-glib --with-gobject --with-cairo --with-icu --with-coretext --with-graphite2
       - run: make -j4
       - run: make check || .ci/fail.sh

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -22,15 +22,16 @@ jobs:
       - run: make -j4
       - run: make check || .ci/fail.sh
 
-  macos-10.14.3-aat-fonts:
+  macos-10.14.4-aat-fonts:
     macos:
-      xcode: "10.2.0"
+      xcode: "11.0.0"
     steps:
       - checkout
       - run: HOMEBREW_NO_AUTO_UPDATE=1 brew install wget autoconf automake libtool pkg-config ragel freetype glib cairo icu4c graphite2
       - run: export PKG_CONFIG_PATH="/usr/local/opt/icu4c/lib/pkgconfig:/usr/local/opt/libffi/lib/pkgconfig" && ./autogen.sh --with-freetype --with-glib --with-gobject --with-cairo --with-icu --with-coretext --with-graphite2
       - run: make -j4
       - run: make check || .ci/fail.sh
+      - run: cmake -Bbuild -H. -DHB_HAVE_CORETEXT=1 && make -Cbuild
 
   distcheck:
     docker:
@@ -306,7 +307,7 @@ workflows:
       # macOS
       - macos-10.12.6-aat-fonts
       - macos-10.13.6-aat-fonts
-      - macos-10.14.3-aat-fonts
+      - macos-10.14.4-aat-fonts
 
       # both autotools and cmake
       - distcheck

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -31,7 +31,7 @@ jobs:
       - run: export PKG_CONFIG_PATH="/usr/local/opt/icu4c/lib/pkgconfig:/usr/local/opt/libffi/lib/pkgconfig" && ./autogen.sh --with-freetype --with-glib --with-gobject --with-cairo --with-icu --with-coretext --with-graphite2
       - run: make -j4
       - run: make check || .ci/fail.sh
-      - run: cmake -Bbuild -H. -DHB_HAVE_CORETEXT=1 && make -Cbuild
+      - run: cmake -Bbuild -H. -DHB_HAVE_CORETEXT=1 -DHB_BUILD_TESTS=0 && cmake --build build
 
   distcheck:
     docker:

--- a/src/hb-aat-layout-bsln-table.hh
+++ b/src/hb-aat-layout-bsln-table.hh
@@ -82,7 +82,7 @@ struct BaselineTableFormat2Part
   }
 
   protected:
-  GlyphID	stdGlyph;	/* The specific glyph index number in this
+  HBGlyphID	stdGlyph;	/* The specific glyph index number in this
 				 * font that is used to set the baseline values.
 				 * This is the standard glyph.
 				 * This glyph must contain a set of control points
@@ -105,7 +105,7 @@ struct BaselineTableFormat3Part
   }
 
   protected:
-  GlyphID	stdGlyph;	/* ditto */
+  HBGlyphID	stdGlyph;	/* ditto */
   HBUINT16	ctlPoints[32];	/* ditto */
   Lookup<HBUINT16>
 		lookupTable;	/* Lookup table that maps glyphs to their

--- a/src/hb-aat-layout-common.hh
+++ b/src/hb-aat-layout-common.hh
@@ -93,8 +93,8 @@ struct LookupSegmentSingle
     return_trace (c->check_struct (this) && value.sanitize (c, base));
   }
 
-  GlyphID	last;		/* Last GlyphID in this segment */
-  GlyphID	first;		/* First GlyphID in this segment */
+  HBGlyphID	last;		/* Last GlyphID in this segment */
+  HBGlyphID	first;		/* First GlyphID in this segment */
   T		value;		/* The lookup value (only one) */
   public:
   DEFINE_SIZE_STATIC (4 + T::static_size);
@@ -162,8 +162,8 @@ struct LookupSegmentArray
 		  valuesZ.sanitize (c, base, last - first + 1, hb_forward<Ts> (ds)...));
   }
 
-  GlyphID	last;		/* Last GlyphID in this segment */
-  GlyphID	first;		/* First GlyphID in this segment */
+  HBGlyphID	last;		/* Last GlyphID in this segment */
+  HBGlyphID	first;		/* First GlyphID in this segment */
   NNOffsetTo<UnsizedArrayOf<T>>
 		valuesZ;	/* A 16-bit offset from the start of
 				 * the table to the data. */
@@ -222,7 +222,7 @@ struct LookupSingle
     return_trace (c->check_struct (this) && value.sanitize (c, base));
   }
 
-  GlyphID	glyph;		/* Last GlyphID */
+  HBGlyphID	glyph;		/* Last GlyphID */
   T		value;		/* The lookup value (only one) */
   public:
   DEFINE_SIZE_STATIC (2 + T::static_size);
@@ -284,7 +284,7 @@ struct LookupFormat8
 
   protected:
   HBUINT16	format;		/* Format identifier--format = 8 */
-  GlyphID	firstGlyph;	/* First glyph index included in the trimmed array. */
+  HBGlyphID	firstGlyph;	/* First glyph index included in the trimmed array. */
   HBUINT16	glyphCount;	/* Total number of glyphs (equivalent to the last
 				 * glyph minus the value of firstGlyph plus 1). */
   UnsizedArrayOf<T>
@@ -326,7 +326,7 @@ struct LookupFormat10
   protected:
   HBUINT16	format;		/* Format identifier--format = 8 */
   HBUINT16	valueSize;	/* Byte size of each value. */
-  GlyphID	firstGlyph;	/* First glyph index included in the trimmed array. */
+  HBGlyphID	firstGlyph;	/* First glyph index included in the trimmed array. */
   HBUINT16	glyphCount;	/* Total number of glyphs (equivalent to the last
 				 * glyph minus the value of firstGlyph plus 1). */
   UnsizedArrayOf<HBUINT8>
@@ -658,7 +658,7 @@ struct ClassTable
     return_trace (c->check_struct (this) && classArray.sanitize (c));
   }
   protected:
-  GlyphID		firstGlyph;	/* First glyph index included in the trimmed array. */
+  HBGlyphID		firstGlyph;	/* First glyph index included in the trimmed array. */
   ArrayOf<HBUCHAR>	classArray;	/* The class codes (indexed by glyph index minus
 					 * firstGlyph). */
   public:

--- a/src/hb-aat-layout-just-table.hh
+++ b/src/hb-aat-layout-just-table.hh
@@ -100,7 +100,7 @@ struct UnconditionalAddGlyphAction
   protected:
   ActionSubrecordHeader
 		header;
-  GlyphID	addGlyph;	/* Glyph that should be added if the distance factor
+  HBGlyphID	addGlyph;	/* Glyph that should be added if the distance factor
 				 * is growing. */
 
   public:
@@ -121,11 +121,11 @@ struct ConditionalAddGlyphAction
   HBFixed 	substThreshold; /* Distance growth factor (in ems) at which
 				 * this glyph is replaced and the growth factor
 				 * recalculated. */
-  GlyphID 	addGlyph; 	/* Glyph to be added as kashida. If this value is
+  HBGlyphID 	addGlyph; 	/* Glyph to be added as kashida. If this value is
 				 * 0xFFFF, no extra glyph will be added. Note that
 				 * generally when a glyph is added, justification
 				 * will need to be redone. */
-  GlyphID 	substGlyph; 	/* Glyph to be substituted for this glyph if the
+  HBGlyphID 	substGlyph; 	/* Glyph to be substituted for this glyph if the
 				 * growth factor equals or exceeds the value of
 				 * substThreshold. */
   public:
@@ -170,7 +170,7 @@ struct RepeatedAddGlyphAction
   ActionSubrecordHeader
 		header;
   HBUINT16 	flags;		/* Currently unused; set to 0. */
-  GlyphID 	glyph;		/* Glyph that should be added if the distance factor
+  HBGlyphID 	glyph;		/* Glyph that should be added if the distance factor
 				 * is growing. */
   public:
   DEFINE_SIZE_STATIC (10);

--- a/src/hb-aat-layout-kerx-table.hh
+++ b/src/hb-aat-layout-kerx-table.hh
@@ -82,8 +82,8 @@ struct KernPair
   }
 
   protected:
-  GlyphID	left;
-  GlyphID	right;
+  HBGlyphID	left;
+  HBGlyphID	right;
   FWORD		value;
   public:
   DEFINE_SIZE_STATIC (6);

--- a/src/hb-aat-layout-morx-table.hh
+++ b/src/hb-aat-layout-morx-table.hh
@@ -240,21 +240,21 @@ struct ContextualSubtable
       if (buffer->idx == buffer->len && !mark_set)
 	return;
 
-      const GlyphID *replacement;
+      const HBGlyphID *replacement;
 
       replacement = nullptr;
       if (Types::extended)
       {
 	if (entry.data.markIndex != 0xFFFF)
 	{
-	  const Lookup<GlyphID> &lookup = subs[entry.data.markIndex];
+	  const Lookup<HBGlyphID> &lookup = subs[entry.data.markIndex];
 	  replacement = lookup.get_value (buffer->info[mark].codepoint, driver->num_glyphs);
 	}
       }
       else
       {
 	unsigned int offset = entry.data.markIndex + buffer->info[mark].codepoint;
-	const UnsizedArrayOf<GlyphID> &subs_old = (const UnsizedArrayOf<GlyphID> &) subs;
+	const UnsizedArrayOf<HBGlyphID> &subs_old = (const UnsizedArrayOf<HBGlyphID> &) subs;
 	replacement = &subs_old[Types::wordOffsetToIndex (offset, table, subs_old.arrayZ)];
 	if (!replacement->sanitize (&c->sanitizer) || !*replacement)
 	  replacement = nullptr;
@@ -272,14 +272,14 @@ struct ContextualSubtable
       {
 	if (entry.data.currentIndex != 0xFFFF)
 	{
-	  const Lookup<GlyphID> &lookup = subs[entry.data.currentIndex];
+	  const Lookup<HBGlyphID> &lookup = subs[entry.data.currentIndex];
 	  replacement = lookup.get_value (buffer->info[idx].codepoint, driver->num_glyphs);
 	}
       }
       else
       {
 	unsigned int offset = entry.data.currentIndex + buffer->info[idx].codepoint;
-	const UnsizedArrayOf<GlyphID> &subs_old = (const UnsizedArrayOf<GlyphID> &) subs;
+	const UnsizedArrayOf<HBGlyphID> &subs_old = (const UnsizedArrayOf<HBGlyphID> &) subs;
 	replacement = &subs_old[Types::wordOffsetToIndex (offset, table, subs_old.arrayZ)];
 	if (!replacement->sanitize (&c->sanitizer) || !*replacement)
 	  replacement = nullptr;
@@ -304,7 +304,7 @@ struct ContextualSubtable
     bool mark_set;
     unsigned int mark;
     const ContextualSubtable *table;
-    const UnsizedOffsetListOf<Lookup<GlyphID>, HBUINT, false> &subs;
+    const UnsizedOffsetListOf<Lookup<HBGlyphID>, HBUINT, false> &subs;
   };
 
   bool apply (hb_aat_apply_context_t *c) const
@@ -348,7 +348,7 @@ struct ContextualSubtable
   protected:
   StateTable<Types, EntryData>
 		machine;
-  NNOffsetTo<UnsizedOffsetListOf<Lookup<GlyphID>, HBUINT, false>, HBUINT>
+  NNOffsetTo<UnsizedOffsetListOf<Lookup<HBGlyphID>, HBUINT, false>, HBUINT>
 		substitutionTables;
   public:
   DEFINE_SIZE_STATIC (20);
@@ -520,7 +520,7 @@ struct LigatureSubtable
 	  if (action & (LigActionStore | LigActionLast))
 	  {
 	    ligature_idx = Types::offsetToIndex (ligature_idx, table, ligature.arrayZ);
-	    const GlyphID &ligatureData = ligature[ligature_idx];
+	    const HBGlyphID &ligatureData = ligature[ligature_idx];
 	    if (unlikely (!ligatureData.sanitize (&c->sanitizer))) break;
 	    hb_codepoint_t lig = ligatureData;
 
@@ -554,7 +554,7 @@ struct LigatureSubtable
     const LigatureSubtable *table;
     const UnsizedArrayOf<HBUINT32> &ligAction;
     const UnsizedArrayOf<HBUINT16> &component;
-    const UnsizedArrayOf<GlyphID> &ligature;
+    const UnsizedArrayOf<HBGlyphID> &ligature;
     unsigned int match_length;
     unsigned int match_positions[HB_MAX_CONTEXT_LENGTH];
   };
@@ -586,7 +586,7 @@ struct LigatureSubtable
 		ligAction;	/* Offset to the ligature action table. */
   NNOffsetTo<UnsizedArrayOf<HBUINT16>, HBUINT>
 		component;	/* Offset to the component table. */
-  NNOffsetTo<UnsizedArrayOf<GlyphID>, HBUINT>
+  NNOffsetTo<UnsizedArrayOf<HBGlyphID>, HBUINT>
 		ligature;	/* Offset to the actual ligature lists. */
   public:
   DEFINE_SIZE_STATIC (28);
@@ -606,7 +606,7 @@ struct NoncontextualSubtable
     unsigned int count = c->buffer->len;
     for (unsigned int i = 0; i < count; i++)
     {
-      const GlyphID *replacement = substitute.get_value (info[i].codepoint, num_glyphs);
+      const HBGlyphID *replacement = substitute.get_value (info[i].codepoint, num_glyphs);
       if (replacement)
       {
 	info[i].codepoint = *replacement;
@@ -624,7 +624,7 @@ struct NoncontextualSubtable
   }
 
   protected:
-  Lookup<GlyphID>	substitute;
+  Lookup<HBGlyphID>	substitute;
   public:
   DEFINE_SIZE_MIN (2);
 };
@@ -726,7 +726,7 @@ struct InsertionSubtable
       {
 	unsigned int count = (flags & MarkedInsertCount);
 	unsigned int start = entry.data.markedInsertIndex;
-	const GlyphID *glyphs = &insertionAction[start];
+	const HBGlyphID *glyphs = &insertionAction[start];
 	if (unlikely (!c->sanitizer.check_array (glyphs, count))) count = 0;
 
 	bool before = flags & MarkedInsertBefore;
@@ -754,7 +754,7 @@ struct InsertionSubtable
       {
 	unsigned int count = (flags & CurrentInsertCount) >> 5;
 	unsigned int start = entry.data.currentInsertIndex;
-	const GlyphID *glyphs = &insertionAction[start];
+	const HBGlyphID *glyphs = &insertionAction[start];
 	if (unlikely (!c->sanitizer.check_array (glyphs, count))) count = 0;
 
 	bool before = flags & CurrentInsertBefore;
@@ -793,7 +793,7 @@ struct InsertionSubtable
     private:
     hb_aat_apply_context_t *c;
     unsigned int mark;
-    const UnsizedArrayOf<GlyphID> &insertionAction;
+    const UnsizedArrayOf<HBGlyphID> &insertionAction;
   };
 
   bool apply (hb_aat_apply_context_t *c) const
@@ -819,7 +819,7 @@ struct InsertionSubtable
   protected:
   StateTable<Types, EntryData>
 		machine;
-  NNOffsetTo<UnsizedArrayOf<GlyphID>, HBUINT>
+  NNOffsetTo<UnsizedArrayOf<HBGlyphID>, HBUINT>
 		insertionAction;	/* Byte offset from stateHeader to the start of
 					 * the insertion glyph table. */
   public:

--- a/src/hb-coretext.cc
+++ b/src/hb-coretext.cc
@@ -302,7 +302,7 @@ _hb_coretext_shaper_font_data_create (hb_font_t *font)
   if (unlikely (!face_data)) return nullptr;
   CGFontRef cg_font = (CGFontRef) (const void *) face->data.coretext;
 
-  CGFloat font_size = font->ptem <= 0.f ? HB_CORETEXT_DEFAULT_FONT_SIZE : font->ptem;
+  CGFloat font_size = (CGFloat) (font->ptem <= 0.f ? HB_CORETEXT_DEFAULT_FONT_SIZE : font->ptem);
   CTFontRef ct_font = create_ct_font (cg_font, font_size);
 
   if (unlikely (!ct_font))
@@ -327,7 +327,7 @@ retry:
   const hb_coretext_font_data_t *data = font->data.coretext;
   if (unlikely (!data)) return nullptr;
 
-  if (fabs (CTFontGetSize ((CTFontRef) data) - font->ptem) > .5)
+  if (fabs (CTFontGetSize ((CTFontRef) data) - (CGFloat) font->ptem) > .5)
   {
     /* XXX-MT-bug
      * Note that evaluating condition above can be dangerous if another thread

--- a/src/hb-open-type.hh
+++ b/src/hb-open-type.hh
@@ -163,9 +163,9 @@ struct Tag : HBUINT32
 };
 
 /* Glyph index number, same as uint16 (length = 16 bits) */
-struct GlyphID : HBUINT16
+struct HBGlyphID : HBUINT16
 {
-  GlyphID& operator = (uint16_t i) { HBUINT16::operator= (i); return *this; }
+  HBGlyphID& operator = (uint16_t i) { HBUINT16::operator= (i); return *this; }
 };
 
 /* Script/language-system/feature index */

--- a/src/hb-ot-cmap-table.hh
+++ b/src/hb-ot-cmap-table.hh
@@ -498,7 +498,7 @@ struct CmapSubtableTrimmed
   UINT		length;		/* Byte length of this subtable. */
   UINT		language;	/* Ignore. */
   UINT		startCharCode;	/* First character code covered. */
-  ArrayOf<GlyphID, UINT>
+  ArrayOf<HBGlyphID, UINT>
 		glyphIdArray;	/* Array of glyph index values for character
 				 * codes in the range. */
   public:
@@ -694,7 +694,7 @@ struct UVSMapping
   }
 
   HBUINT24	unicodeValue;	/* Base Unicode value of the UVS */
-  GlyphID	glyphID;	/* Glyph ID of the UVS */
+  HBGlyphID	glyphID;	/* Glyph ID of the UVS */
   public:
   DEFINE_SIZE_STATIC (5);
 };

--- a/src/hb-ot-color-cbdt-table.hh
+++ b/src/hb-ot-color-cbdt-table.hh
@@ -226,8 +226,8 @@ struct IndexSubtableRecord
 						   offset, length, format);
   }
 
-  GlyphID			firstGlyphIndex;
-  GlyphID			lastGlyphIndex;
+  HBGlyphID			firstGlyphIndex;
+  HBGlyphID			lastGlyphIndex;
   LOffsetTo<IndexSubtable>	offsetToSubtable;
   public:
   DEFINE_SIZE_STATIC(8);
@@ -290,8 +290,8 @@ struct BitmapSizeTable
   HBUINT32		colorRef;
   SBitLineMetrics	horizontal;
   SBitLineMetrics	vertical;
-  GlyphID		startGlyphIndex;
-  GlyphID		endGlyphIndex;
+  HBGlyphID		startGlyphIndex;
+  HBGlyphID		endGlyphIndex;
   HBUINT8		ppemX;
   HBUINT8		ppemY;
   HBUINT8		bitDepth;

--- a/src/hb-ot-color-colr-table.hh
+++ b/src/hb-ot-color-colr-table.hh
@@ -48,7 +48,7 @@ struct LayerRecord
   }
 
   protected:
-  GlyphID	glyphId;	/* Glyph ID of layer glyph */
+  HBGlyphID	glyphId;	/* Glyph ID of layer glyph */
   Index		colorIdx;	/* Index value to use with a
 				 * selected color palette.
 				 * An index value of 0xFFFF
@@ -75,7 +75,7 @@ struct BaseGlyphRecord
   }
 
   public:
-  GlyphID	glyphId;	/* Glyph ID of reference glyph */
+  HBGlyphID	glyphId;	/* Glyph ID of reference glyph */
   HBUINT16	firstLayerIdx;	/* Index (from beginning of
 				 * the Layer Records) to the
 				 * layer record. There will be

--- a/src/hb-ot-glyf-table.hh
+++ b/src/hb-ot-glyf-table.hh
@@ -330,7 +330,7 @@ struct glyf
     };
 
     HBUINT16 flags;
-    GlyphID  glyphIndex;
+    HBGlyphID  glyphIndex;
 
     unsigned int get_size () const
     {

--- a/src/hb-ot-layout-base-table.hh
+++ b/src/hb-ot-layout-base-table.hh
@@ -73,7 +73,7 @@ struct BaseCoordFormat2
   protected:
   HBUINT16	format;		/* Format identifier--format = 2 */
   FWORD		coordinate;	/* X or Y value, in design units */
-  GlyphID	referenceGlyph;	/* Glyph ID of control glyph */
+  HBGlyphID	referenceGlyph;	/* Glyph ID of control glyph */
   HBUINT16	coordPoint;	/* Index of contour point on the
 				 * reference glyph */
   public:

--- a/src/hb-ot-layout-common.hh
+++ b/src/hb-ot-layout-common.hh
@@ -172,8 +172,8 @@ struct RangeRecord
   bool add_coverage (set_t *glyphs) const
   { return glyphs->add_range (start, end); }
 
-  GlyphID	start;		/* First GlyphID in the range */
-  GlyphID	end;		/* Last GlyphID in the range */
+  HBGlyphID	start;		/* First GlyphID in the range */
+  HBGlyphID	end;		/* Last GlyphID in the range */
   HBUINT16	value;		/* Value */
   public:
   DEFINE_SIZE_STATIC (6);
@@ -856,7 +856,7 @@ struct CoverageFormat1
 
   protected:
   HBUINT16	coverageFormat;	/* Format identifier--format = 1 */
-  SortedArrayOf<GlyphID>
+  SortedArrayOf<HBGlyphID>
 		glyphArray;	/* Array of GlyphIDs--in numerical order */
   public:
   DEFINE_SIZE_ARRAY (4, glyphArray);
@@ -1196,7 +1196,7 @@ struct Coverage
  */
 
 static inline void ClassDef_serialize (hb_serialize_context_t *c,
-				       hb_array_t<const GlyphID> glyphs,
+				       hb_array_t<const HBGlyphID> glyphs,
 				       hb_array_t<const HBUINT16> klasses);
 
 struct ClassDefFormat1
@@ -1210,7 +1210,7 @@ struct ClassDefFormat1
   }
 
   bool serialize (hb_serialize_context_t *c,
-		  hb_array_t<const GlyphID> glyphs,
+		  hb_array_t<const HBGlyphID> glyphs,
 		  hb_array_t<const HBUINT16> klasses)
   {
     TRACE_SERIALIZE (this);
@@ -1241,7 +1241,7 @@ struct ClassDefFormat1
     TRACE_SUBSET (this);
     const hb_set_t &glyphset = *c->plan->glyphset ();
     const hb_map_t &glyph_map = *c->plan->glyph_map;
-    hb_sorted_vector_t<GlyphID> glyphs;
+    hb_sorted_vector_t<HBGlyphID> glyphs;
     hb_vector_t<HBUINT16> klasses;
 
     hb_codepoint_t start = startGlyph;
@@ -1328,7 +1328,7 @@ struct ClassDefFormat1
 
   protected:
   HBUINT16	classFormat;	/* Format identifier--format = 1 */
-  GlyphID	startGlyph;	/* First GlyphID of the classValueArray */
+  HBGlyphID	startGlyph;	/* First GlyphID of the classValueArray */
   ArrayOf<HBUINT16>
 		classValue;	/* Array of Class Values--one per GlyphID */
   public:
@@ -1346,7 +1346,7 @@ struct ClassDefFormat2
   }
 
   bool serialize (hb_serialize_context_t *c,
-		  hb_array_t<const GlyphID> glyphs,
+		  hb_array_t<const HBGlyphID> glyphs,
 		  hb_array_t<const HBUINT16> klasses)
   {
     TRACE_SERIALIZE (this);
@@ -1390,7 +1390,7 @@ struct ClassDefFormat2
     TRACE_SUBSET (this);
     const hb_set_t &glyphset = *c->plan->glyphset ();
     const hb_map_t &glyph_map = *c->plan->glyph_map;
-    hb_vector_t<GlyphID> glyphs;
+    hb_vector_t<HBGlyphID> glyphs;
     hb_vector_t<HBUINT16> klasses;
 
     unsigned int count = rangeRecord.len;
@@ -1506,7 +1506,7 @@ struct ClassDef
   }
 
   bool serialize (hb_serialize_context_t *c,
-		  hb_array_t<const GlyphID> glyphs,
+		  hb_array_t<const HBGlyphID> glyphs,
 		  hb_array_t<const HBUINT16> klasses)
   {
     TRACE_SERIALIZE (this);
@@ -1611,7 +1611,7 @@ struct ClassDef
 };
 
 static inline void ClassDef_serialize (hb_serialize_context_t *c,
-				       hb_array_t<const GlyphID> glyphs,
+				       hb_array_t<const HBGlyphID> glyphs,
 				       hb_array_t<const HBUINT16> klasses)
 { c->start_embed<ClassDef> ()->serialize (c, glyphs, klasses); }
 

--- a/src/hb-ot-layout-gpos-table.hh
+++ b/src/hb-ot-layout-gpos-table.hh
@@ -750,7 +750,7 @@ struct PairValueRecord
   friend struct PairSet;
 
   protected:
-  GlyphID	secondGlyph;		/* GlyphID of second glyph in the
+  HBGlyphID	secondGlyph;		/* GlyphID of second glyph in the
 					 * pair--first glyph is listed in the
 					 * Coverage table */
   ValueRecord	values;			/* Positioning data for the first glyph

--- a/src/hb-ot-layout-gsub-table.hh
+++ b/src/hb-ot-layout-gsub-table.hh
@@ -212,7 +212,7 @@ struct SingleSubstFormat2
     + hb_zip (this+coverage, substitute)
     | hb_filter (glyphset, hb_first)
     | hb_filter (glyphset, hb_second)
-    | hb_map_retains_sorting ([&] (hb_pair_t<hb_codepoint_t, const GlyphID &> p) -> hb_codepoint_pair_t
+    | hb_map_retains_sorting ([&] (hb_pair_t<hb_codepoint_t, const HBGlyphID &> p) -> hb_codepoint_pair_t
 			      { return hb_pair (glyph_map[p.first], glyph_map[p.second]); })
     ;
 
@@ -232,7 +232,7 @@ struct SingleSubstFormat2
   OffsetTo<Coverage>
 		coverage;		/* Offset to Coverage table--from
 					 * beginning of Substitution table */
-  ArrayOf<GlyphID>
+  ArrayOf<HBGlyphID>
 		substitute;		/* Array of substitute
 					 * GlyphIDs--ordered by Coverage Index */
   public:
@@ -374,7 +374,7 @@ struct Sequence
   }
 
   protected:
-  ArrayOf<GlyphID>
+  ArrayOf<HBGlyphID>
 		substitute;		/* String of GlyphIDs to substitute */
   public:
   DEFINE_SIZE_ARRAY (2, substitute);
@@ -421,9 +421,9 @@ struct MultipleSubstFormat1
   }
 
   bool serialize (hb_serialize_context_t *c,
-		  hb_sorted_array_t<const GlyphID> glyphs,
+		  hb_sorted_array_t<const HBGlyphID> glyphs,
 		  hb_array_t<const unsigned int> substitute_len_list,
-		  hb_array_t<const GlyphID> substitute_glyphs_list)
+		  hb_array_t<const HBGlyphID> substitute_glyphs_list)
   {
     TRACE_SERIALIZE (this);
     if (unlikely (!c->extend_min (*this))) return_trace (false);
@@ -496,9 +496,9 @@ struct MultipleSubstFormat1
 struct MultipleSubst
 {
   bool serialize (hb_serialize_context_t *c,
-		  hb_sorted_array_t<const GlyphID> glyphs,
+		  hb_sorted_array_t<const HBGlyphID> glyphs,
 		  hb_array_t<const unsigned int> substitute_len_list,
-		  hb_array_t<const GlyphID> substitute_glyphs_list)
+		  hb_array_t<const HBGlyphID> substitute_glyphs_list)
   {
     TRACE_SERIALIZE (this);
     if (unlikely (!c->extend_min (u.format))) return_trace (false);
@@ -597,7 +597,7 @@ struct AlternateSet
   }
 
   protected:
-  ArrayOf<GlyphID>
+  ArrayOf<HBGlyphID>
 		alternates;		/* Array of alternate GlyphIDs--in
 					 * arbitrary order */
   public:
@@ -644,9 +644,9 @@ struct AlternateSubstFormat1
   }
 
   bool serialize (hb_serialize_context_t *c,
-		  hb_sorted_array_t<const GlyphID> glyphs,
+		  hb_sorted_array_t<const HBGlyphID> glyphs,
 		  hb_array_t<const unsigned int> alternate_len_list,
-		  hb_array_t<const GlyphID> alternate_glyphs_list)
+		  hb_array_t<const HBGlyphID> alternate_glyphs_list)
   {
     TRACE_SERIALIZE (this);
     if (unlikely (!c->extend_min (*this))) return_trace (false);
@@ -719,9 +719,9 @@ struct AlternateSubstFormat1
 struct AlternateSubst
 {
   bool serialize (hb_serialize_context_t *c,
-		  hb_sorted_array_t<const GlyphID> glyphs,
+		  hb_sorted_array_t<const HBGlyphID> glyphs,
 		  hb_array_t<const unsigned int> alternate_len_list,
-		  hb_array_t<const GlyphID> alternate_glyphs_list)
+		  hb_array_t<const HBGlyphID> alternate_glyphs_list)
   {
     TRACE_SERIALIZE (this);
     if (unlikely (!c->extend_min (u.format))) return_trace (false);
@@ -860,8 +860,8 @@ struct Ligature
   }
 
   protected:
-  GlyphID	ligGlyph;		/* GlyphID of ligature to substitute */
-  HeadlessArrayOf<GlyphID>
+  HBGlyphID	ligGlyph;		/* GlyphID of ligature to substitute */
+  HeadlessArrayOf<HBGlyphID>
 		component;		/* Array of component GlyphIDs--start
 					 * with the second  component--ordered
 					 * in writing direction */
@@ -921,9 +921,9 @@ struct LigatureSet
   }
 
   bool serialize (hb_serialize_context_t *c,
-		  hb_array_t<const GlyphID> ligatures,
+		  hb_array_t<const HBGlyphID> ligatures,
 		  hb_array_t<const unsigned int> component_count_list,
-		  hb_array_t<const GlyphID> &component_list /* Starting from second for each ligature */)
+		  hb_array_t<const HBGlyphID> &component_list /* Starting from second for each ligature */)
   {
     TRACE_SERIALIZE (this);
     if (unlikely (!c->extend_min (*this))) return_trace (false);
@@ -1038,11 +1038,11 @@ struct LigatureSubstFormat1
   }
 
   bool serialize (hb_serialize_context_t *c,
-		  hb_sorted_array_t<const GlyphID> first_glyphs,
+		  hb_sorted_array_t<const HBGlyphID> first_glyphs,
 		  hb_array_t<const unsigned int> ligature_per_first_glyph_count_list,
-		  hb_array_t<const GlyphID> ligatures_list,
+		  hb_array_t<const HBGlyphID> ligatures_list,
 		  hb_array_t<const unsigned int> component_count_list,
-		  hb_array_t<const GlyphID> component_list /* Starting from second for each ligature */)
+		  hb_array_t<const HBGlyphID> component_list /* Starting from second for each ligature */)
   {
     TRACE_SERIALIZE (this);
     if (unlikely (!c->extend_min (*this))) return_trace (false);
@@ -1118,11 +1118,11 @@ struct LigatureSubstFormat1
 struct LigatureSubst
 {
   bool serialize (hb_serialize_context_t *c,
-		  hb_sorted_array_t<const GlyphID> first_glyphs,
+		  hb_sorted_array_t<const HBGlyphID> first_glyphs,
 		  hb_array_t<const unsigned int> ligature_per_first_glyph_count_list,
-		  hb_array_t<const GlyphID> ligatures_list,
+		  hb_array_t<const HBGlyphID> ligatures_list,
 		  hb_array_t<const unsigned int> component_count_list,
-		  hb_array_t<const GlyphID> component_list /* Starting from second for each ligature */)
+		  hb_array_t<const HBGlyphID> component_list /* Starting from second for each ligature */)
   {
     TRACE_SERIALIZE (this);
     if (unlikely (!c->extend_min (u.format))) return_trace (false);
@@ -1199,7 +1199,7 @@ struct ReverseChainSingleSubstFormat1
     if (!intersects (c->glyphs)) return;
 
     const OffsetArrayOf<Coverage> &lookahead = StructAfter<OffsetArrayOf<Coverage>> (backtrack);
-    const ArrayOf<GlyphID> &substitute = StructAfter<ArrayOf<GlyphID>> (lookahead);
+    const ArrayOf<HBGlyphID> &substitute = StructAfter<ArrayOf<HBGlyphID>> (lookahead);
 
     + hb_zip (this+coverage, substitute)
     | hb_filter (*c->glyphs, hb_first)
@@ -1223,7 +1223,7 @@ struct ReverseChainSingleSubstFormat1
     for (unsigned int i = 0; i < count; i++)
       if (unlikely (!(this+lookahead[i]).add_coverage (c->after))) return;
 
-    const ArrayOf<GlyphID> &substitute = StructAfter<ArrayOf<GlyphID>> (lookahead);
+    const ArrayOf<HBGlyphID> &substitute = StructAfter<ArrayOf<HBGlyphID>> (lookahead);
     count = substitute.len;
     c->output->add_array (substitute.arrayZ, substitute.len);
   }
@@ -1243,7 +1243,7 @@ struct ReverseChainSingleSubstFormat1
     if (likely (index == NOT_COVERED)) return_trace (false);
 
     const OffsetArrayOf<Coverage> &lookahead = StructAfter<OffsetArrayOf<Coverage>> (backtrack);
-    const ArrayOf<GlyphID> &substitute = StructAfter<ArrayOf<GlyphID>> (lookahead);
+    const ArrayOf<HBGlyphID> &substitute = StructAfter<ArrayOf<HBGlyphID>> (lookahead);
 
   unsigned int start_index = 0, end_index = 0;
     if (match_backtrack (c,
@@ -1281,7 +1281,7 @@ struct ReverseChainSingleSubstFormat1
     const OffsetArrayOf<Coverage> &lookahead = StructAfter<OffsetArrayOf<Coverage>> (backtrack);
     if (!lookahead.sanitize (c, this))
       return_trace (false);
-    const ArrayOf<GlyphID> &substitute = StructAfter<ArrayOf<GlyphID>> (lookahead);
+    const ArrayOf<HBGlyphID> &substitute = StructAfter<ArrayOf<HBGlyphID>> (lookahead);
     return_trace (substitute.sanitize (c));
   }
 
@@ -1298,7 +1298,7 @@ struct ReverseChainSingleSubstFormat1
 		lookaheadX;		/* Array of coverage tables
 					 * in lookahead sequence, in glyph
 					 * sequence order */
-  ArrayOf<GlyphID>
+  ArrayOf<HBGlyphID>
 		substituteX;		/* Array of substitute
 					 * GlyphIDs--ordered by Coverage Index */
   public:
@@ -1453,8 +1453,8 @@ struct SubstLookup : Lookup
 
   bool serialize_single (hb_serialize_context_t *c,
 			 uint32_t lookup_props,
-			 hb_sorted_array_t<const GlyphID> glyphs,
-			 hb_array_t<const GlyphID> substitutes)
+			 hb_sorted_array_t<const HBGlyphID> glyphs,
+			 hb_array_t<const HBGlyphID> substitutes)
   {
     TRACE_SERIALIZE (this);
     if (unlikely (!Lookup::serialize (c, SubTable::Single, lookup_props, 1))) return_trace (false);
@@ -1464,9 +1464,9 @@ struct SubstLookup : Lookup
 
   bool serialize_multiple (hb_serialize_context_t *c,
 			   uint32_t lookup_props,
-			   hb_sorted_array_t<const GlyphID> glyphs,
+			   hb_sorted_array_t<const HBGlyphID> glyphs,
 			   hb_array_t<const unsigned int> substitute_len_list,
-			   hb_array_t<const GlyphID> substitute_glyphs_list)
+			   hb_array_t<const HBGlyphID> substitute_glyphs_list)
   {
     TRACE_SERIALIZE (this);
     if (unlikely (!Lookup::serialize (c, SubTable::Multiple, lookup_props, 1))) return_trace (false);
@@ -1479,9 +1479,9 @@ struct SubstLookup : Lookup
 
   bool serialize_alternate (hb_serialize_context_t *c,
 			    uint32_t lookup_props,
-			    hb_sorted_array_t<const GlyphID> glyphs,
+			    hb_sorted_array_t<const HBGlyphID> glyphs,
 			    hb_array_t<const unsigned int> alternate_len_list,
-			    hb_array_t<const GlyphID> alternate_glyphs_list)
+			    hb_array_t<const HBGlyphID> alternate_glyphs_list)
   {
     TRACE_SERIALIZE (this);
     if (unlikely (!Lookup::serialize (c, SubTable::Alternate, lookup_props, 1))) return_trace (false);
@@ -1494,11 +1494,11 @@ struct SubstLookup : Lookup
 
   bool serialize_ligature (hb_serialize_context_t *c,
 			   uint32_t lookup_props,
-			   hb_sorted_array_t<const GlyphID> first_glyphs,
+			   hb_sorted_array_t<const HBGlyphID> first_glyphs,
 			   hb_array_t<const unsigned int> ligature_per_first_glyph_count_list,
-			   hb_array_t<const GlyphID> ligatures_list,
+			   hb_array_t<const HBGlyphID> ligatures_list,
 			   hb_array_t<const unsigned int> component_count_list,
-			   hb_array_t<const GlyphID> component_list /* Starting from second for each ligature */)
+			   hb_array_t<const HBGlyphID> component_list /* Starting from second for each ligature */)
   {
     TRACE_SERIALIZE (this);
     if (unlikely (!Lookup::serialize (c, SubTable::Ligature, lookup_props, 1))) return_trace (false);

--- a/src/hb-ot-layout-jstf-table.hh
+++ b/src/hb-ot-layout-jstf-table.hh
@@ -136,7 +136,7 @@ struct JstfLangSys : OffsetListOf<JstfPriority>
  * ExtenderGlyphs -- Extender Glyph Table
  */
 
-typedef SortedArrayOf<GlyphID> ExtenderGlyphs;
+typedef SortedArrayOf<HBGlyphID> ExtenderGlyphs;
 
 
 /*

--- a/src/hb-ot-math-table.hh
+++ b/src/hb-ot-math-table.hh
@@ -423,7 +423,7 @@ struct MathGlyphVariantRecord
   }
 
   protected:
-  GlyphID variantGlyph;       /* Glyph ID for the variant. */
+  HBGlyphID variantGlyph;       /* Glyph ID for the variant. */
   HBUINT16  advanceMeasurement; /* Advance width/height, in design units, of the
 				 * variant, in the direction of requested
 				 * glyph extension. */
@@ -471,7 +471,7 @@ struct MathGlyphPartRecord
   }
 
   protected:
-  GlyphID   glyph;		  /* Glyph ID for the part. */
+  HBGlyphID   glyph;		  /* Glyph ID for the part. */
   HBUINT16    startConnectorLength; /* Advance width/ height of the straight bar
 				   * connector material, in design units, is at
 				   * the beginning of the glyph, in the

--- a/src/hb-ot-name-table.hh
+++ b/src/hb-ot-name-table.hh
@@ -330,6 +330,9 @@ struct name
   DEFINE_SIZE_ARRAY (6, nameRecordZ);
 };
 
+#undef entry_index
+#undef entry_score
+
 struct name_accelerator_t : name::accelerator_t {};
 
 } /* namespace OT */

--- a/src/hb-ot-shape-complex-arabic-fallback.hh
+++ b/src/hb-ot-shape-complex-arabic-fallback.hh
@@ -49,8 +49,8 @@ arabic_fallback_synthesize_lookup_single (const hb_ot_shape_plan_t *plan HB_UNUS
 					  hb_font_t *font,
 					  unsigned int feature_index)
 {
-  OT::GlyphID glyphs[SHAPING_TABLE_LAST - SHAPING_TABLE_FIRST + 1];
-  OT::GlyphID substitutes[SHAPING_TABLE_LAST - SHAPING_TABLE_FIRST + 1];
+  OT::HBGlyphID glyphs[SHAPING_TABLE_LAST - SHAPING_TABLE_FIRST + 1];
+  OT::HBGlyphID substitutes[SHAPING_TABLE_LAST - SHAPING_TABLE_FIRST + 1];
   unsigned int num_glyphs = 0;
 
   /* Populate arrays */
@@ -78,7 +78,7 @@ arabic_fallback_synthesize_lookup_single (const hb_ot_shape_plan_t *plan HB_UNUS
   /* Bubble-sort or something equally good!
    * May not be good-enough for presidential candidate interviews, but good-enough for us... */
   hb_stable_sort (&glyphs[0], num_glyphs,
-		  (int(*)(const OT::HBUINT16*, const OT::HBUINT16 *)) OT::GlyphID::cmp,
+		  (int(*)(const OT::HBUINT16*, const OT::HBUINT16 *)) OT::HBGlyphID::cmp,
 		  &substitutes[0]);
 
 
@@ -99,15 +99,15 @@ static OT::SubstLookup *
 arabic_fallback_synthesize_lookup_ligature (const hb_ot_shape_plan_t *plan HB_UNUSED,
 					    hb_font_t *font)
 {
-  OT::GlyphID first_glyphs[ARRAY_LENGTH_CONST (ligature_table)];
+  OT::HBGlyphID first_glyphs[ARRAY_LENGTH_CONST (ligature_table)];
   unsigned int first_glyphs_indirection[ARRAY_LENGTH_CONST (ligature_table)];
   unsigned int ligature_per_first_glyph_count_list[ARRAY_LENGTH_CONST (first_glyphs)];
   unsigned int num_first_glyphs = 0;
 
   /* We know that all our ligatures are 2-component */
-  OT::GlyphID ligature_list[ARRAY_LENGTH_CONST (first_glyphs) * ARRAY_LENGTH_CONST(ligature_table[0].ligatures)];
+  OT::HBGlyphID ligature_list[ARRAY_LENGTH_CONST (first_glyphs) * ARRAY_LENGTH_CONST(ligature_table[0].ligatures)];
   unsigned int component_count_list[ARRAY_LENGTH_CONST (ligature_list)];
-  OT::GlyphID component_list[ARRAY_LENGTH_CONST (ligature_list) * 1/* One extra component per ligature */];
+  OT::HBGlyphID component_list[ARRAY_LENGTH_CONST (ligature_list) * 1/* One extra component per ligature */];
   unsigned int num_ligatures = 0;
 
   /* Populate arrays */
@@ -125,7 +125,7 @@ arabic_fallback_synthesize_lookup_ligature (const hb_ot_shape_plan_t *plan HB_UN
     num_first_glyphs++;
   }
   hb_stable_sort (&first_glyphs[0], num_first_glyphs,
-		  (int(*)(const OT::HBUINT16*, const OT::HBUINT16 *)) OT::GlyphID::cmp,
+		  (int(*)(const OT::HBUINT16*, const OT::HBUINT16 *)) OT::HBGlyphID::cmp,
 		  &first_glyphs_indirection[0]);
 
   /* Now that the first-glyphs are sorted, walk again, populate ligatures. */

--- a/src/hb-ot-vorg-table.hh
+++ b/src/hb-ot-vorg-table.hh
@@ -48,7 +48,7 @@ struct VertOriginMetric
   }
 
   public:
-  GlyphID	glyph;
+  HBGlyphID	glyph;
   FWORD		vertOriginY;
 
   public:

